tests/integration/integration_test_case.py

```diff
@@ -268,3 +280,21 @@ def assertRegexUrl(self, regex):
             self.assertRegex(text=self.last_url, expected_regex=regex)
         else:
             self.fail('last_url is invalid')
+
+
+def decode_flask_cookie(cookie):
+    """Decode a Flask cookie."""
+    compressed = False
+    payload = cookie
+
+    if payload.startswith('.'):
+        compressed = True
+        payload = payload[1:]
+
+    data = payload.split('.')[0]
+
+    data = base64_decode(data)
+    if compressed:
```

Comment:
If our cookie is always compressed then we should always try and decompress, therefore we can tell if we suddenly stop compressign it